src/components/Table/TableBase.tsx

```diff
@@ -184,10 +186,13 @@ export class TableBase<T> extends React.Component<
 		}
 
 		if (
-			this.props.checkedItems &&
-			prevProps.checkedItems !== this.props.checkedItems
+			(this.props.checkedItems &&
+				prevProps.checkedItems !== this.props.checkedItems) ||
+			(this.props.data && prevProps.data !== this.props.data)
 		) {
-			this.setRowSelection(this.props.checkedItems);
+			if (this.props.checkedItems) {
+				this.setRowSelection(this.props.checkedItems);
+			}
```

Comment:
```
if (page !== undefined && prevProps.page !== page) {
			this.setState({ page });
		}
```
There's no `page` prop, not sure if you noted that from an older change I made 😅 

WRT the data, it changes when the filters change (for example adding a filter might remove some entities). With that said, I do not believe this is necessary anymore because now if the data changes (or rather the filters update, adding or removing data), we remove all selections, just like the behavior of filter application in gmail.